src/test/rustdoc/deref-const-fn.rs

```diff
@@ -0,0 +1,37 @@
+// This test ensures that the const methods from Deref aren't const anymore.
+// For more information, take a look at https://github.com/rust-lang/rust/issues/67792.
+#![crate_name = "foo"]
+
+#![feature(staged_api)]
+
+#![stable(feature = "rust1", since = "1.0.0")]
+
+// @has 'foo/struct.Bar.html'
+#[stable(feature = "rust1", since = "1.0.0")]
+pub struct Bar;
+
+impl Bar {
+    // @has - '//*[@id="method.len"]' 'pub const fn len(&self) -> usize'
+    // @has - '//*[@id="method.len"]//span[@class="since"]' '1.0.0 (const: 1.0.0)'
+    #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_const_stable(feature = "rust1", since = "1.0.0")]
+    pub const fn len(&self) -> usize { 0 }
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+pub struct Foo {
+    value: Bar,
+}
+
+// @has 'foo/struct.Foo.html'
+// @has - '//*[@id="method.len"]' 'pub fn len(&self) -> usize'
+// @!has - '//*[@id="method.len"]//span[@class="since"]' '1.0.0'
```

Comment:
Hmm, shouldn't stability be shown? Or I guess it depends on when the Deref impl was added too.

Weirdly, https://doc.rust-lang.org/nightly/std/string/struct.String.html#deref-methods-str shows regular and const stability, but https://doc.rust-lang.org/nightly/std/path/struct.PathBuf.html#deref-methods-Path shows neither. Something fishy's going on with re-exports, since https://doc.rust-lang.org/nightly/alloc/string/struct.String.html#deref-methods-str shows stability for some methods but not all and is missing many methods that the std version has. (I think I opened an issue for part of this at some point.)

Reply:
Since you already opened the issue, I'll mark this discussion as resolved.

Reply:
No, the issue I opened was just about the methods shown being different. But I guess this bug is unrelated to this PR?

Reply:
I thought `1.0.0` without the const version is hidden. Maybe that's why? If there is a stable const version, then we show `1.0.0 (const: 1.xx.0)` or `const: unstable`.